library/src/data/index.js

```diff
@@ -41,20 +41,265 @@ window.CoNDeT.data = {
     },
 
     saveToFile: function (data) {
-      var element = document.createElement('a');
+      var element = document.createElement("a");
       element.setAttribute(
-        'href',
-        'data:text/plain;charset=utf-8,' +
+        "href",
+        "data:text/plain;charset=utf-8," +
           encodeURIComponent(JSON.stringify(data))
       );
-      element.setAttribute('download', 'CoNDeT.json');
+      element.setAttribute("download", "CoNDeT.json");
 
-      element.style.display = 'none';
+      element.style.display = "none";
       document.body.appendChild(element); //Required on firefox
 
       element.click();
 
       element.remove();
     },
   },
+
+  StateModifier: (function () {
+    constructor = function (state) {
+      this.stateManager = state;
+      var self = this;
+      state.subscribe(function (newState) {
+        self.state = newState;
+      });
+    };
+
+    moveTable = function (tableId, newCoordinates) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+      editTable.coordinates = {
+        x: newCoordinates.x,
+        y: newCoordinates.y,
+      };
+
+      this.stateManager.setState(newState);
+    };
+
+    createTable = function (newData) {
+      var newState = clone(this.state);
+      newState.push(newData);
+      this.stateManager.setState(newState);
+    };
+
+    editName = function (tableId, name) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      if (name != null) {
+        editTable.name = name;
+      }
```

Comment:
Dając negację warunku na samym początku można uniknąć niepotrzebnego update-a tym samym stanem. + kod w moim subiektywnym odczuciu jest lepszy
```
editName = function (tableId, name) {
  if (name == null) return;
  var newState = clone(this.state);
  var editTable = getTableWithId(newState, tableId);
  editTable.name = name;
  this.stateManager.setState(newState);
}
``` 
Zastanawiam się też, co jeżeli `tableId` jest `null`? Biorąc pod uwagę, że tylko my wewnętrznie będziemy używać tej metody (API biblioteki nie będzie ich udostępniać) to nie wiem czy takie wybiórcze sprawdzani ma sens. Odpuścił bym to przyjmując, że dane są dobre. Normalnie TypeScript rozwiązuje ten problem ciągłego sprawdzania typów, no ale wymagania projektu sprawiają, że nie możemy go użyć. Więc myślę, że finalnie najlepiej będzie zostawić coś takiego:
```
editName = function (tableId, name) {
  var newState = clone(this.state);
  var editTable = getTableWithId(newState, tableId);
  editTable.name = name;
  this.stateManager.setState(newState);
}
``` 
I analogicznie pozostałe funkcje jak ta